src/_guides/language/common-prob.md

```diff
@@ -0,0 +1,473 @@
+but the code assumes it returns the subtype CanvasElement
+where `context2D` is defined.
+The `canvas` field is declared as `var` which, in classic Dart,
+types it as `dynamic` and silences all errors.
+Strong mode Dart infers `canvas` to be an Element.
+
+Fix this error with an explicit type declaration:
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+[[highlight]]CanvasElement[[/highlight]] canvas = querySelector("canvas");
+canvas.context2D;
+{% endprettify %}
+</div>
+
+If you actually want a dynamic type, specify `dynamic`:
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+[[highlight]]dynamic[[/highlight]] canvas = querySelector("canvas");
+canvas.context2D;
+{% endprettify %}
+</div>
+
+<hr>
+
+<a name="invalid-method-override"></a>
+### Invalid method override
+
+**Error:** <code>Invalid override. The type of &lt;<em>type</em>&gt; is not a subtype of &lt;<em>type</em>&gt;.</code>
+
+These errors typically occur when a subclass tightens up a method's
+parameter types by specifying a subclass of the original class.
+
+<aside class="alert alert-info" markdown="1">
+**Note:** This warning can also occur when a generic subclass neglects
+to specify a type. For more information, see
+[Missing type arguments](#missing-type-arguments).
+</aside>
+
+**Fix:** Widen the types in the method's parameter list.
+The subclass's method should accept every
+object that the superclass's method takes.
+
+In the following example, the parameters in the `add()` method
+are changed from `num` to `int`, a subtype of `num`.
+This code passes static analysis in classic Dart,
+but is unsafe and fails analysis in strong mode Dart.
+
+<div class="fails-sa" markdown="1">
+{% prettify dart %}
+abstract class NumberAdder {
+  num add(num a, num b);
+}
+
+class IntAdder extends NumberAdder {
+  int add([[highlight]]int[[/highlight]] a, [[highlight]]int[[/highlight]] b) => a + b;
+}
+{% endprettify %}
+</div>
+
+Consider the following scenario where floating
+point values are passed to an IntAdder:
+
+{% prettify dart %}
+NumberAdder adder = new IntAdder(); // Upcast
+adder.add([[highlight]]1.2[[/highlight]], [[highlight]]3.4[[/highlight]]);                // Kaboom!
+{% endprettify %}
+
+If the override were allowed, this code would crash at runtime.
+
+Fix this error by widening the types in the subclass:
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+abstract class NumberAdder {
+  num add(num a, num b);
+}
+
+class IntAdder extends NumberAdder {
+  num add([[highlight]]num[[/highlight]] a, [[highlight]]num[[/highlight]] b) => a + b;
+}
+{% endprettify %}
+</div>
+
+For more information, see [Use proper input parameter types when overriding methods](/guides/language/sound-dart#use-proper-input-parameter-types-when-overriding-methods-).
+
+<aside class="alert alert-info" markdown="1">
+**Note:**
+Some coding patterns require this functionality.
+If you have a valid reason to do this,
+use the `@checked` annotation. For example:
```

Comment:
This is being renamed to `covariant` (without `@`) in the next release of Dart, hopefully. See: https://github.com/dart-lang/sdk/issues/27486

You might want to just not mention it at all here, since it should be rare that people should use this.